packages/typescript-estree/src/create-program/createProjectProgram.ts

```diff
@@ -34,65 +34,82 @@ function createProjectProgram(
 ): ASTAndProgram | undefined {
   log('Creating project program for: %s', extra.filePath);
 
-  const astAndProgram = firstDefined(
-    getProgramsForProjects(code, extra.filePath, extra),
-    currentProgram => getAstFromProgram(currentProgram, extra),
+  const programsForProjects = getProgramsForProjects(
+    code,
+    extra.filePath,
+    extra,
+  );
+  const astAndProgram = firstDefined(programsForProjects, currentProgram =>
+    getAstFromProgram(currentProgram, extra),
   );
 
-  if (!astAndProgram && !createDefaultProgram) {
-    // the file was either not matched within the tsconfig, or the extension wasn't expected
-    const errorLines = [
-      '"parserOptions.project" has been set for @typescript-eslint/parser.',
-      `The file does not match your project config: ${path.relative(
-        extra.tsconfigRootDir || process.cwd(),
-        extra.filePath,
-      )}.`,
-    ];
-    let hasMatchedAnError = false;
+  // The file was either matched within the tsconfig, or we allow creating a default program
+  if (astAndProgram || createDefaultProgram) {
+    return astAndProgram;
+  }
 
-    const extraFileExtensions = extra.extraFileExtensions || [];
+  const describeFilePath = (filePath: string): string =>
+    path.relative(extra.tsconfigRootDir || process.cwd(), filePath);
 
-    extraFileExtensions.forEach(extraExtension => {
-      if (!extraExtension.startsWith('.')) {
-        errorLines.push(
-          `Found unexpected extension "${extraExtension}" specified with the "extraFileExtensions" option. Did you mean ".${extraExtension}"?`,
-        );
-      }
-      if (DEFAULT_EXTRA_FILE_EXTENSIONS.includes(extraExtension)) {
-        errorLines.push(
-          `You unnecessarily included the extension "${extraExtension}" with the "extraFileExtensions" option. This extension is already handled by the parser by default.`,
-        );
-      }
-    });
+  const describedFilePath = describeFilePath(extra.filePath);
+  const relativeProjects = extra.projects.map(describeFilePath);
+  const describedPrograms =
+    relativeProjects.length === 1
+      ? relativeProjects[0]
+      : `\n${relativeProjects.map(project => `- ${project}`).join('\n')}`;
```

Comment:
doesn't look like there's a test case for this multiple project config case.
probably worth adding one!

Reply:
There is: `throws when non of multiple projects include the file`